### PR TITLE
Add changelogs for 1.10.0

### PR DESCRIPTION
**Pull request by MarcosDY**
Update change log with 1.10.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,37 @@
...
+- PSAT node attestor to cross check the audience fields (#5142)
+- Events-based cache to handle out of order events (#5071)
+
+### Deprecated
+
+- `x509_svid_cache_max_size` and `disable_lru_cache` in agent configuration (#5150)
+
+### Removed
+
+- The deprecated `disable_reattest_to_renew` configurable (#5217)
```
Comment (amartinezfayo) [inline CHANGELOG.md:28]:
```suggestion
- The deprecated `disable_reattest_to_renew` agent configurable (#5217)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+- The deprecated `use_msi` flag from `azure_msi` plugin (#5209)
```
Comment (amartinezfayo) [inline CHANGELOG.md:29]:
```suggestion
- The deprecated `use_msi` flag from `azure_msi` NodeAttestor plugin (#5209)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,37 @@
...
+- The deprecated `key_metadata_file` and `use_msi` configurables (#5207)
```
Comment (amartinezfayo) [inline CHANGELOG.md:30]:
```suggestion
- The deprecated `key_metadata_file` and `use_msi` configurables from the `azure_key_vault` server KeyManager (#5207)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,37 @@
...
+- The deprecated `exclude_sn_from_ca_subject` configurable (#5203)
```
Comment (amartinezfayo) [inline CHANGELOG.md:31]:
```suggestion
- The deprecated `exclude_sn_from_ca_subject` server configurable (#5203)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,37 @@
...
+- Agent no longer cleans up deprecated bundle/SVID files (#5205)
```
Comment (amartinezfayo) [inline CHANGELOG.md:32]:
```suggestion
- Agent no longer cleans up deprecated bundle and SVID files (#5205)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,37 @@
...
+- The journal file is no longer stored (#5202)
```
Comment (amartinezfayo) [inline CHANGELOG.md:33]:
```suggestion
- The CA journal file is no longer stored on disk, and existing CA journal files are cleaned up (#5202)
```

**Review by amartinezfayo — COMMENTED** [review]
Thank you @MarcosDY for putting this together! I have some suggestions.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,37 @@
 # Changelog
 
+## [1.10.0] - 2024-06-24
+
+### Added
+
+- Reconfiguration support to plugins using `plugin_data_file` configurable (#5166)
```
Comment (azdagron) [inline CHANGELOG.md:7]:
```suggestion
- Plugin reconfiguration support using the `plugin_data_file` configurable (#5166)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,37 @@
...
+
+### Added
+
+- Reconfiguration support to plugins using `plugin_data_file` configurable (#5166) 
+
+### Changed
+
+- SPIRE Server and OIDC provider images to use non root users (#4967, #5227) 
+- K8s workload attestor to no longer fail when a cluster is not configured (#5216)
+- Reattestable plugins to fail on renew (#5204)
```
[the remark this answers is not in the document]
Comment (azdagron) [reply CHANGELOG.md:13]: Maybe "are required" instead of "are enforced"?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,37 @@
...
+- Small documentation improvements (#5181, #5189)
+- SPIRE Agent to reattest when evicting (#4991)
```
[the remark this answers is not in the document]
Comment (azdagron) [reply CHANGELOG.md:15]: Maybe `Evicted agents that support reattestation can now reattest without being restarted (#4991)`?

File: CHANGELOG.md (hunk 2 above)
Comment (azdagron) [reply CHANGELOG.md:29]: from *the* `azure_msi`

**Review by azdagron — APPROVED** [review]

**Review by amartinezfayo — APPROVED** [review]

**Pull request merged**